test/scalecube-cluster/cluster.test.js

```diff
@@ -0,0 +1,113 @@
+import { LogicalCluster as Cluster } from 'src/scalecube-cluster/LogicalCluster';
+import { from } from 'rxjs/observable/from';
+import 'rxjs/add/operator/zip';
+import 'rxjs/add/operator/do';
+import 'rxjs/add/operator/elementAt';
+
+describe('Cluster suite', () => {
+  const createClusters = () => {
+
+    const clusterA = new Cluster();
+    const clusterB = new Cluster();
+    const clusterC = new Cluster();
+
+
+
+    return {clusterA, clusterB, clusterC};
+  };
+
+  it('When clusterA join clusterB and B join C all cluster should be on all clusters', () => {
+    const {clusterA, clusterB, clusterC} = createClusters();
+
+    clusterA.join(clusterB);
+    clusterB.join(clusterC);
+
+    expect(clusterA.members()).toEqual([ clusterA, clusterB, clusterC ]);
+    expect(clusterB.members()).toEqual([ clusterB, clusterA, clusterC ]);
+    expect(clusterC.members()).toEqual([ clusterC, clusterB, clusterA ]);
+  });
+  it('When clusterA shutdown clusterB and C should not have clusterA and remove message should be send', () => {
+    const {clusterA, clusterB, clusterC} = createClusters();
+    expect.assertions(8);
+
+    clusterA.join(clusterB);
+    clusterB.join(clusterC);
+
+      [clusterA,clusterB,clusterC].forEach(cluster =>
+          cluster
+              .listenMembership()
+              .do((msg) => {
```

Comment:
maybe alternative can be
.onNext((msg) => {
.on((msg) => {

Reply:
rxjs don't have onNext and on

Reply:
@idanilt Can **subscribe()** be considered as **onNext**?